pkg/pgxconn/pgx_conn.go

```diff
@@ -37,21 +47,63 @@ type connImpl struct {
 	Conn *pgxpool.Pool
 }
 
+type pgxRows struct {
+	pgx.Rows
+	sqlQuery  string
+	args      []interface{}
+	startTime time.Time
+	loggged   bool
+}
+
 func (p *connImpl) Close() {
 	conn := p.Conn
 	p.Conn = nil
 	conn.Close()
 }
 
-func (p *connImpl) Exec(ctx context.Context, sql string, arguments ...interface{}) (pgconn.CommandTag, error) {
-	return p.Conn.Exec(ctx, sql, arguments...)
+func (p *pgxRows) Next() bool {
+	if !p.loggged {
+		p.loggged = true
+		LogQueryStats(p.sqlQuery, p.startTime, p.args)()
+	}
+	return p.Rows.Next()
+}
```

Comment:
We have to cann p.Rows.Next before logging to make sure the timing is right
```suggestion
func (p *pgxRows) Next() bool { 
        //The query fetch is async and happens on the first call to next. 
        //so to get timing right, we log timing after first Next() call.
	if !p.loggged {
		p.loggged = true
	         res := p.Rows.Next()
		LogQueryStats(p.sqlQuery, p.startTime, p.args)()
		return res
	}
	return p.Rows.Next()
}
```